src/auto-evo/simulation/SimulationCache.cs

```diff
@@ -43,7 +44,8 @@ public class SimulationCache
     private readonly Dictionary<(TweakedProcess, IBiomeConditions), ProcessSpeedInformation> cachedProcessSpeeds =
         new();
 
-    private readonly Dictionary<MicrobeSpecies, (float, float, float, float)> cachedPredationToolsRawScores = new();
+    private readonly Dictionary<MicrobeSpecies, (float, float, float, float, HashSet<string>)>
```

Comment:
I think that adding a HashSet here is starting to make this way too bloated to keep using this current architecture...

I think the enzymes need to be saved differently. Maybe splitting the data into multiple parts? Like one general predation tools score and then relying on caching the enzyme scores per predated-on species (in predationScores).